README.md

```diff
@@ -0,0 +1,132 @@
+# Ballerina-Transformer Tools
+
+
+A transformer is a general purpose interface to apply a function to a set of data points. The output may contain data points generated by adding, removing or altering some. Ballerina-Transformer is also a such tool that provides the following capabilities.
+
+1. Validate if a package is a transformer package
+2. Generate the Ballerina service code for the transformer functions
+
+## Validator and Generator
+### Transformer Package Validator
+Ballerina transformer tools can validate a Ballerina package if that comply with the constraints of a Ballerina transformer package. A Ballerina package to be treated as a transformer package it should comply for the below-mentioned constraints.
+1. The package should not contain any entry-points _(ie: main function or services)_
+2. The package should have one or more expression-bodied functions (transformer-function) with public and isolated qualifier.
+3. The package can contain other functions which can be used at util functions (cannot be public).
+4. The package can contain records
+5. The package in not allowed to have class definitions or listener definitions
+6. Expression bodied functions in the package is only allowed to have public qualifier
+7. The package is not allowed to have annotations.
+8. The public isolated expression bodied functions (transformer-functions) are allowed to have serializable types as parameter type and return type
+
+To activate the transformer-tools in a Ballerina package to validate the package, the tools has to be imported. Once, it is imported, when we build the Ballerina transformer package, it would validate against the constraints.
+```ballerina
+import ballerinax/transformer as _;
+```
+
+### Ballerina Service Generator
+Once a Ballerina package is validated, the transformer-tools will generate a service where it would allow the transformer-functions to be utilized through http.
```

Comment:
```suggestion
Once a Ballerina package is validated, the transformer tools will generate a service where it would allow the transformer functions to be consumed through REST APIs. The parameters of the transformer function has to be passed as a JSON payload.
```